### PR TITLE
PAN-OS - Add CRUD commands for EDL, refresh-edl, register/unregister-ip-tag commands

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/16545
fixes: https://github.com/demisto/etc/issues/16546
fixes: https://github.com/demisto/etc/issues/16269
fixes: https://github.com/demisto/etc/issues/16645
reference: https://github.com/demisto/etc/issues/16584

## Description
support for the EDL object in PAN-OS

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation https://github.com/demisto/etc/issues/16542
- [x] Code Review

### CODE REVIEW

**Comment by bakatzir** [comment]
Please see new commits, fixing also naming of product, add support for refresh-edl and DAG handling.
updated tests and documentation accordingly.

File: Integrations/integration-Panorama.yml [hunk 1]
```diff
@@ -2299,6 +2324,396 @@ script:
...
+        edl_type = ''.join(edl_prev['type'].keys())
+        edl_output = {'Name': edl_name}
+        params = {
+            'action': 'edit',
+            'type': 'config',
+            'key': API_KEY,
+        }
+
+        if element_to_change == 'url':
+            params['xpath'] = XPATH_OBJECTS + "external-list/entry[@name='" + edl_name + "']/type/" + edl_type + "/" + element_to_change
```
Comment (yuvalbenshalom) [inline Integrations/integration-Panorama.yml]: if you use the same code in all the lines(2506,2512,2518...) Think it is more readable to create it and put it in a ve before the switch case

File: Integrations/integration-Panorama.yml [hunk 2]
```diff
@@ -2299,6 +2324,396 @@ script:
...
+        ''' Register IPs to a Tag '''
+        tag = demisto.args()['tag']
+        ips = argToList(demisto.args()['IPs'])
+        persistent = demisto.args()['persistent'] if 'persistent' in demisto.args() else '1'
+
+        result = panorama_register_ip_tag(tag, ips, persistent)
+
+        registered_ip = {}
+        # update context only if IPs are persistent
+        if persistent == '1':
```
Comment (yuvalbenshalom) [inline Integrations/integration-Panorama.yml]: shouldn't this be a boolean?

File: Integrations/integration-Panorama.yml [hunk 3]
```diff
@@ -3111,6 +3552,151 @@ script:
       description: URL filter rule name
       type: string
     description: Deletes a URL filtering rule
+  - name: panorama-list-edls
+    arguments: []
+    outputs:
+    - contextPath: Panorama.EDLs.Name
```
Comment (yuvalbenshalom) [inline Integrations/integration-Panorama.yml]: why EDLs and not just EDL? seems bit confusing.

File: Integrations/integration-Panorama.yml [hunk 4]
```diff
@@ -3492,7 +4078,42 @@ script:
...
+  - name: panorama-register-ip-tag
+    arguments:
+    - name: tag
+      required: true
+      description: Tag to register IPs to
+    - name: IPs
+      required: true
+      description: IPs to register
+      isArray: true
+    - name: persistent
```
Comment (yuvalbenshalom) [inline Integrations/integration-Panorama.yml]: why 0,1 and not true false? is that something from the API

File: Integrations/integration-Panorama.yml [hunk 5]
```diff
@@ -176,6 +197,9 @@ script:
             else:
                 return_error('Request Failed.\n' + str(jresult['response']))
 
+        if 'response' in jresult and '@code' in jresult['response'] and jresult['response']['@code'] in PAN_OS_ERROR_DICT:
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Integrations/integration-Panorama.yml]: sometimes, dependeing on api call, the @code key is not present. so I thought we shouldn't catch unknown @code values.

File: Integrations/integration-Panorama.yml (hunk 2 above)
Comment (bakatzir) [reply Integrations/integration-Panorama.yml]: no, it's a numeric value according to the api reference.

File: Integrations/integration-Panorama.yml (hunk 4 above)
Comment (bakatzir) [reply Integrations/integration-Panorama.yml]: yes. i can twik the args and context values to bollean if that's more readable

File: Integrations/integration-Panorama.yml [hunk 6]
```diff
@@ -131,6 +131,27 @@ script:
         'target': 'Target'
     }
 
+    PAN_OS_ERROR_DICT = {
```
[the remark this answers is not in the document]
Comment (bakatzir) [reply Integrations/integration-Panorama.yml]: approved.

**Comment by bakatzir** [comment]
@yuvalbenshalom added errors of existing ips in tag as entry.note:
Screen Shot 2019-04-22 at 11 08 31

**Comment by yuvalbenshalom** [comment]
@bakatzir please make sure @kirbles19 review the changes in the yml and the error code dict

**Comment by bakatzir** [comment]
@yuvalbenshalom  he reviewed all error code dict strings.
yml changes i can wait for him to finish, but prefer to do al string changes in a new PR.

**Comment by yuvalbenshalom** [comment]
@bakatzir our new workflow is to review the yml before merge as much as possible.

**Pull request merged**